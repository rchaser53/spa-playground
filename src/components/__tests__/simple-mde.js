import Vue from 'vue'
import * as VueServerRenderer from 'vue-server-renderer'
import SimpleMde from '../simple-mde/simple-mde.vue'

const renderer = VueServerRenderer.createRenderer()

const render = (vm) => {
  return new Promise((resolve, reject) => {
    renderer.renderToString(vm, (err, str) => {
      if (err) {
        return reject(err)
      }
      resolve(str)
    })
  })
}

test('render simple-mde', () => {
  const vm = new Vue({
    el: document.createElement('div'),
    render: (h) => {
      h('div', SimpleMde)
    }
  })

	expect(vm).toMatchSnapshot()

})
