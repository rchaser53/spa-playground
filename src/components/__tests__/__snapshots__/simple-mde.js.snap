// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render simple-mde 1`] = `
Vue {
  "$attrs": Object {},
  "$children": Array [],
  "$createElement": [Function],
  "$el": <!---->,
  "$listeners": Object {},
  "$options": Object {
    "_base": [Function],
    "_parentElm": null,
    "_refElm": null,
    "components": Object {},
    "directives": Object {},
    "el": <div />,
    "filters": Object {},
    "render": [Function],
  },
  "$parent": undefined,
  "$refs": Object {},
  "$root": [Circular],
  "$scopedSlots": Object {},
  "$slots": Object {},
  "$vnode": undefined,
  "_c": [Function],
  "_data": Object {},
  "_directInactive": false,
  "_events": Object {},
  "_hasHookEvent": false,
  "_inactive": null,
  "_isBeingDestroyed": false,
  "_isDestroyed": false,
  "_isMounted": true,
  "_isVue": true,
  "_renderProxy": Vue {
    "$attrs": Object {},
    "$children": Array [],
    "$createElement": [Function],
    "$el": <!---->,
    "$listeners": Object {},
    "$options": Object {
      "_base": [Function],
      "_parentElm": null,
      "_refElm": null,
      "components": Object {},
      "directives": Object {},
      "el": <div />,
      "filters": Object {},
      "render": [Function],
    },
    "$parent": undefined,
    "$refs": Object {},
    "$root": [Circular],
    "$scopedSlots": Object {},
    "$slots": Object {},
    "$vnode": undefined,
    "_c": [Function],
    "_data": Object {},
    "_directInactive": false,
    "_events": Object {},
    "_hasHookEvent": false,
    "_inactive": null,
    "_isBeingDestroyed": false,
    "_isDestroyed": false,
    "_isMounted": true,
    "_isVue": true,
    "_renderProxy": [Circular],
    "_self": [Circular],
    "_staticTrees": null,
    "_uid": 0,
    "_vnode": VNode {
      "asyncFactory": undefined,
      "asyncMeta": undefined,
      "children": undefined,
      "componentInstance": undefined,
      "componentOptions": undefined,
      "context": undefined,
      "data": undefined,
      "elm": <!---->,
      "fnContext": undefined,
      "fnOptions": undefined,
      "fnScopeId": undefined,
      "isAsyncPlaceholder": false,
      "isCloned": false,
      "isComment": true,
      "isOnce": false,
      "isRootInsert": true,
      "isStatic": false,
      "key": undefined,
      "ns": undefined,
      "parent": undefined,
      "raw": false,
      "tag": undefined,
      "text": "",
    },
    "_watcher": Watcher {
      "active": true,
      "cb": [Function],
      "deep": false,
      "depIds": Set {},
      "deps": Array [],
      "dirty": false,
      "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
      "getter": [Function],
      "id": 1,
      "lazy": false,
      "newDepIds": Set {},
      "newDeps": Array [],
      "sync": false,
      "user": false,
      "value": undefined,
      "vm": [Circular],
    },
    "_watchers": Array [
      Watcher {
        "active": true,
        "cb": [Function],
        "deep": false,
        "depIds": Set {},
        "deps": Array [],
        "dirty": false,
        "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
        "getter": [Function],
        "id": 1,
        "lazy": false,
        "newDepIds": Set {},
        "newDeps": Array [],
        "sync": false,
        "user": false,
        "value": undefined,
        "vm": [Circular],
      },
    ],
  },
  "_self": [Circular],
  "_staticTrees": null,
  "_uid": 0,
  "_vnode": VNode {
    "asyncFactory": undefined,
    "asyncMeta": undefined,
    "children": undefined,
    "componentInstance": undefined,
    "componentOptions": undefined,
    "context": undefined,
    "data": undefined,
    "elm": <!---->,
    "fnContext": undefined,
    "fnOptions": undefined,
    "fnScopeId": undefined,
    "isAsyncPlaceholder": false,
    "isCloned": false,
    "isComment": true,
    "isOnce": false,
    "isRootInsert": true,
    "isStatic": false,
    "key": undefined,
    "ns": undefined,
    "parent": undefined,
    "raw": false,
    "tag": undefined,
    "text": "",
  },
  "_watcher": Watcher {
    "active": true,
    "cb": [Function],
    "deep": false,
    "depIds": Set {},
    "deps": Array [],
    "dirty": false,
    "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
    "getter": [Function],
    "id": 1,
    "lazy": false,
    "newDepIds": Set {},
    "newDeps": Array [],
    "sync": false,
    "user": false,
    "value": undefined,
    "vm": [Circular],
  },
  "_watchers": Array [
    Watcher {
      "active": true,
      "cb": [Function],
      "deep": false,
      "depIds": Set {},
      "deps": Array [],
      "dirty": false,
      "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
      "getter": [Function],
      "id": 1,
      "lazy": false,
      "newDepIds": Set {},
      "newDeps": Array [],
      "sync": false,
      "user": false,
      "value": undefined,
      "vm": [Circular],
    },
  ],
}
`;
